Events correlation

lbonn opened the pull request:
So we get a special id from director's targets.json and send it on per-ECU reports.

Not fully tested yet, but I've updated test data to include this field.

patrickvacek commented on tests/test_data/repo/keys/director/private.key:
```diff
@@ -1,27 +1,27 @@
+bcDlbDswH+7ueqPbo33+7cbx5oSUiG+uVZ5pSphCgElMez0EonGNh/e6aXFkZmht
+h0P0wQKBgGtaHs54qwfmCcDgtg4taqGQ3xdyq2136BSNVdy83UbHzsFKu16D568i
+2RY8K+n4ViISHvw5rbZsdQOll8HDQjCFqZ/2ueh8u/dKrKElZjicymFp7bojQ6+R
+C55roLHTQNmgUU6hwtQSwuSn/3Si5NgLoTVP/G0DNg4nqaNQXhGm
```
Did these expire, or what necessitated the change?

patrickvacek replied:
Did these expire, or was it just the addition of the correlation ID that necessitated changing these?

patrickvacek replied:
Did you have to reset the expiration, or were these just changed to include the correlation ID?

patrickvacek commented on tests/test_data/repo/keys/director/private.key:
```diff
@@ -1,27 +1,27 @@
+bcDlbDswH+7ueqPbo33+7cbx5oSUiG+uVZ5pSphCgElMez0EonGNh/e6aXFkZmht
+h0P0wQKBgGtaHs54qwfmCcDgtg4taqGQ3xdyq2136BSNVdy83UbHzsFKu16D568i
+2RY8K+n4ViISHvw5rbZsdQOll8HDQjCFqZ/2ueh8u/dKrKElZjicymFp7bojQ6+R
+C55roLHTQNmgUU6hwtQSwuSn/3Si5NgLoTVP/G0DNg4nqaNQXhGm
```
Did these expire, or was it just the addition of the correlation ID that necessitated changing these?

lbonn replied:
For future reference: only updated for adding the correlation id.

patrickvacek commented on tests/test_data/repo/generate.sh:
```diff
@@ -0,0 +1,60 @@
+akrepo --command addtarget --hwid sec_hwid1 --serial sec_serial1 --filename secondary_firmware.txt
+akrepo --command addtarget --hwid sec_hwid2 --serial sec_serial2 --filename secondary_firmware2.txt
+akrepo --command signtargets
+move_meta multisec
```
Nice! I recently had to figure out how to do this stuff as well for some of the tests, so this is cool to have to make that easier to figure out.

patrickvacek commented:
Jeez, sorry about all the confusing comments. I've tried to remove all the redundant bits.

lbonn commented:
Thanks @patrickvacek, I'll work on adding proper tests. Also CI has been acting weird, will look into that.

lbonn commented:
Ok, I've added a little bit of testing, included in `test_aktualizr`. It's only getting installation events, so download events are not tested.

simao commented on docs/ecu_events.adoc:
````diff
@@ -0,0 +1,177 @@
+```
+{
+        "deviceTime" : "2018-10-25T12:35:42Z",
+        "event" : "{\"signatures\":[{\"keyid\":\"798f4a19aa185f97c3c119e78f251cc30f293b226c3dc1ccd8694f15b1f36d5d\",\"method\":\"rsassa-pss\",\"sig\":\"ck5pu2LPxFYs39d6eDauGseBHWwmHTE6VNJ2Ngs+Iu6zveFjYPf/IU6P/ucBkWC8MnlgswLZeIznXU/+ynSoL95W5wHWYDxKruzTOCYUmXJxTFK+q4cnnfOgvBmoDh35M889vrqFQtsp2SG8P6VIe4iPnb02wbopVcW4FcCLks+3idg8lk15W0Pm3yedkmu7ycJk8ZuY11V8Xrt4at+DCcsK+TD0YiKmbypsBDAkS+lyjYz7aYqyXiL ...
````
why is this one escaped?

lbonn replied:
Yes, that's not great but that's what is going out now. Not sure if it's a regression or the historical behavior.

At a minimum, it could be:

```
"event" : {
  "targets": {
    "signatures": [ ... ]
    ...
  }
}
```

But I think @jerrytrieu said that we don't even care about this field.

Should I make the change anyway?

patrickvacek replied:
That is historical behavior, for better or worse. That was the first event we created for this system and at the time Vladimir didn't seem to care about the details and I didn't put much thought into it. I'd say it is worth making it consistent with the other event reports, though.

jerrytrieu replied:
Yes, let's make this send `correlationId` as the others. Along with `ecu` and `target` filename.

lbonn commented:
Ok I've rebased it. Last item to resolve is the test without correlation id. I'll try to add something which does not add too much duplication.

jerrytrieu commented on docs/ecu_events.adoc:
```diff
@@ -0,0 +1,177 @@
+        "deviceTime" : "2018-10-25T12:35:42Z",
+        "event" :
+        {
+        here-ota:mtus:c017f968-5c7e-4bf6-9492-24a4e90e933b
```
Typo.

jerrytrieu replied:
typo.

lbonn commented:
@jerrytrieu I could add correlation id to `DownloadComplete` but right now it contains info about multiple ecus. Should it be merged with `EcuDownloadCompleted` somehow, so that we only keep per-ecu events? 
Would you then prefer to keep the old name with `version` set to 2?

jerrytrieu commented:
@lbonn I see it's separate events, then new `EcuDownloadComplete` is good, and forget old one

lbonn commented:
@patrickvacek Ok I've added a test for the case without correlation id. Took the opportunity to experiment with ephemeral uptane repository instead of the classic but convoluted `tests/test_data/repo`.

Also removed the old `DownloadComplete` for good!

jerrytrieu commented on docs/ecu_events.adoc:
```diff
@@ -0,0 +1,153 @@
+        "deviceTime" : "2018-10-25T12:35:42Z",
+        "event" :
+        {
+                "correlationId" : "here-ota:mtus:c017f968-5c7e-4bf6-9992-2134ff06432c",
```
Let's use `here-ota:campaigns:....` as example, as `mtus` are going away.
